### ISSUE TITLE
Drop Googles horrible conscrypt

### ISSUE THREAD

**Issue by wltjr** [netty/netty#7189]
I have no idea why this was added to a patch release. Google makes horrible stuff and their repos are a mess. Now netty depends on conscrypt, which in turn depends now on Google's boringssl. Which [they recommend](https://github.com/google/boringssl) others not use.

"BoringSSL is a fork of OpenSSL that is designed to meet Google's needs.

Although BoringSSL is an open source project, it is not intended for general use, as OpenSSL is. We don't recommend that third parties depend upon it. Doing so is likely to be frustrating because there are no guarantees of API or ABI stability."

All furthering stuff from tomcat-native, that netty modifies to netty-tc-native, and now Google messing up further with conscrypt. Its making a real mess and a nightmare for packagers. Even the conscrypt sources google puts on Maven no longer build due to missing constants. That conscrypt generates from a c file, [generate_constants.cc](), Which in of itself is insane. That includes borringssl modified openssl headers. Its a complete mess.

I already dislike Google, their java stuff is horrendous. I would never use any Google Java or other software. This type of packaging madness is making me really hate Java.

Please please use projects with good upstreams. Please do not add dependencies to a project on patch releases. Please ensure outside projects are of release quality and not rapidly changing. None of this would be the case if netty did not use conscypt. Its become problematic since.

This effects both 4.0.x and 4.1.x with both switching to conscrypt at 4.1.10 and 4.0.45. That is really something that should be done on a major or minor release, not a patch.

**Comment by normanmaurer** [comment]
@wltjr its a optional dependency so you are free to just not use it by not put it on your class path. Netty will still work and just either use netty-tcnative if its on the class path or JDK SSL impl.

**Comment by wltjr** [comment]
@normanmaurer maybe optional runtime but not build time. This is regarding my efforts in [packaging Netty](https://github.com/Obsidian-StudiosInc/os-xtoo/tree/master/dev-java/netty-handler) on Gentoo. I do not have a choice in packaging conscrypt. Unless I delete the optional code using it.

A user may still want to, and then I will have to see about packaging boringssl which is a build and runtime dependency of conscrypt. It is NOT like netty-tc-native. That does not require extra packages to build. If boringssl was intended for 3rd parties and had a release that would be one thing. The relationship between conscrypt and boringssl is odd. conscrypt uses a C program to create a NativeConstants class. Which you must compile, then run that program to get the NativeConstants.java. Which is not present in rc9 of conscrypt on maven. You cannot even build it from their sources on maven.

conscrypt needs at minimum the C headers from boringssl to be able to compile the generate_natives program, and then run that to get NativeConstants.java. Which cannot be avoided. Since it seems netty-handler needs at least rc8 or rc9 of conscrypt now.

Its a big mess, easily avoided by not using conscrypt. Google has made a mess of that. Go look into building conscrypt, boringssl, and see if that is really something Netty wants to be connected to, and dependent on. Even if an optional runtime, it is still mandatory build time.

**Comment by normanmaurer** [comment]
@nmittler any idea how to "fix this"...

**Comment by normanmaurer** [comment]
@wltjr btw I think you would be better of to be less "aggressive" with your issue titles etc otherwise people may feel not really motived to help you.  Netty itself receive a lot of contributions from google and we even have google engineers in the core team...

Just my 2 ' cents ....

**Comment by normanmaurer** [comment]
And just to clarify I am not saying there is no issue with how conscrypt is packaged / build and how it is integrated with netty its just not the "nicest" way to report an issue... Anyway let us wait for @nmittler and see if he can help here...

**Comment by wltjr** [comment]
One thought it  netty-handler could be split further, netty-handler-conscrypt etc. Though I really think you all should look into how conscrypt is build. If it was using openssl it would be one thing. Google needs to decide on what they are going to do with boringssl. Both conscrypt and boringssl are Google projects and tightly coupled, as conscrypt needs live headers from boringssl. Thus it really needs to be present and "installed" to build conscrypt. I am not sure why Google cannot contribute to openssl and/or netty-tc-native, and must make a fork of a fork regarding netty-tc-native. Or at least make boringssl usable for others, do a release, etc.

Thank you for your consideration! Conscrypt was not a big deal when I first packaged it using maven sources. Though I was questioning why maven sources vs being able to build directly. Which I am now. I do not like having to constantly touch things I have packaged. I had to some what re-do the conscrypt package. I know I have a considerable amount of work to do still to make conscrypt usable. I had to be creative to package rc9 as it was.

I would like it if Linus Torvalds was a nice guy. I would like it if I did not have frustrations from projects that considered dependency when added. Sorry for the aggressiveness, but it comes from frustration resulting from time loss and the mess.

When it comes to anything relating to Google, nice is out the door. I do not want to take this off topic. I have numerous technical issues with Google. I am a Linux and Java guy, anti Microsoft. But I never hated any company. I absolutely hate Google. Their stuff is a mess, a joke, and I do not use it. I would really question their contributions to Netty.

For example, their Youtube Java APIs and code is HORRIBLE!!! I packaged all that mess. I will never use it. I used JSON simple to do what I needed. I find it interesting I can use say one facebook4j for most all of facebook. While google has a [bunch of garbage](https://github.com/Obsidian-StudiosInc/os-xtoo/tree/master/dev-java) really. Every version of Guava must be slotted,  on Gentoo. Various packages need specific versions etc. Google has destroyed Android. I have issues on maps, chromium, etc. Technical stuff.

I also have personal and business issues with Google, recently delisted from maps after some google certified contributor left me a 1 star review with no feedback. Which Google argued with me and then removed my business entirely. Google is causing me so many problems. I am trying to do what I can daily to avoid them.

I cannot stress enough how much I hate google over technical matters. With how I see them being socially, being a republican. I know I would never fit in at google, be accepted, listened to etc. I guarantee my hostility is less than what they would show towards someone like me. My frustrations and gripes with them are 100% legitimate, technical, and I can back up with facts, links, etc.

Not one of my issues only effects me. Google just does not care. They are always right, everyone else is wrong. Their influence and reach is ever growing, as is the mess they are creating.

Please please please consider this information. I have no ill will towards Google. I am just sick of the problems they cause in my life on so many fronts. They are absolute evil! I have sent them thousands in business in the past as well....

**Comment by dimitarg** [comment]
https://github.com/fantasylandinst/fcop or some of the ideas might be of use.

**Comment by wltjr** [comment]
Something that kills innovation -> [Google I/O 2008 - Open Source Projects and Poisonous People ](https://www.youtube.com/watch?v=-F-3E8pyjFo)

Something that breeds innovation -> [Steve Jobs - Beautiful Polished Rocks](https://www.youtube.com/watch?v=K-Yv-UdsmSo)

Proof
https://www.infoworld.com/article/3109830/linux/why-did-gentoo-linux-fade-into-obscurity.html
http://distrowatch.com/table.php?distribution=exherbo

I always conduct myself professionally. You may not like what I am saying or the manner in which I state it, aggressive, etc. But it is not violating almost any CoC, including the referenced one... Its called work not fun. Many times in life people say things you do not want to hear, or state it in a manner you do not like. That does not mean it is not professional or even insulting, though you may take it as such. Maybe others need to learn people skills to handle people who have a style and point of view that differs with their own. Most anyone who works in a call center is faced with such daily. Do they drive customers away or find means to calm them and find out the cause of the frustration/aggressiveness. Either way, they find a way to deal, or hang up and make it worse. Ever work retail, there is no escaping, short of calling security or the police... Good luck doing that with an irate customer....

Having seen what CoC enforcement does to communities. If you want to stifle innovation and drive people away. Go right ahead and proceed with such. Not to mention inconsistent application of CoC. Which I have seen more than once, as its always up to interpretation. I tend to look more to the root cause vs some sort of punishment to appease others, usually a minority.

Something that maybe of interest -> [Partial Initial Patches to update Zookeeper trunk to Netty 4.x API](https://issues.apache.org/jira/browse/ZOOKEEPER-2399)

There is still legacy stuff in wide use that needs to be updated. While others fork and create a mess of new projects. With no mind to what this does to others, the downstream effect. I am not sure people making this stuff even care about others per the text on boringssl readme.

I do not see how stuff like conscrypt helps others FOSS projects update to current Netty. Zookeeper is still using 3.x from org.jboss not even io.netty...

**Comment by jeffjirsa** [comment]
For someone who claims to harbor no ill will towards google, you sure do love piling on unnecessary commentary that adds nothing new to the (closed) issue. Your assertion that you conduct yourself professionally is laughable. The project marked the issue as closed. If you have nothing new to contribute, no new commentary is needed. Irrelevant rants about CoC belong on your blog, not github.

**Comment by wltjr** [comment]
@jeffjirsa What are you adding that is on topic? Or @dimitarg 

Judging and making comments such as ones conduct is not professional and laughable is a direct insult. That is something covered by CoC. Why I said interpretation, as I am sure @jeffjirsa you think your comments are pertinent to this issue, while calling me out for my comments on an issue I opened. While you proceed to add nothing of value, and seem to violate CoC yourself.

Having a strong dislike even hatred for a business, Google. Is not the same as having ill will. I am not saying I want to see anything happen to Google. Like you all point out my conduct. I want to see Google change its conduct. So does the EU. it will only be a matter of time before other Governments like the US one act. None of that am I talking about, supporting etc. I am not saying anything should happen to Google.

I just have major issues with most all being technical and a few other, relating to my business, etc. 100% legitimate gripes. Given how often Google is being talked about in the news. Many clearly have  growing issue with them. This is all off topic.

Why I tried to bring alls attention to things like the Zookeeper situation. Which I was working on, and the comments who rather bring up CoC, say its a rant needs to go to blog etc. They are not helping there. Maybe coding in other areas, but i am adding things of value...

Why I conclude with saying adding conscrypt to netty does not really help. It will not help any project using older netty move to newer. I was very shocked to see it added as a new dependency for a patch release. Not even a minor version release. Added to both 4.0.x and 4.1.x about the same time. I assume netty users wanted that.
